Add compare function (#3832)
brimdata/zed: docs/language/functions/compare.md

### Function

> &emsp; **compare** &mdash; returns an int comparing two values

Reviewer: ```suggestion
&emsp; **compare** &mdash; returns an integer comparing two values
```

> &emsp; **compare** &mdash; returns an integer comparing two values

Reviewer: ```suggestion
&emsp; **compare** &mdash; return an integer comparing two values
```

### Synopsis

```
compare(a: any, b: any) -> int64
```

### Description

> The _compare_ function returns an int comparing two values. The result with

Reviewer: ```suggestion
The _compare_ function returns an integer comparing two values. The result will
```

be 0 if a is equal to b, +1 if a is greater than b, and -1 if a is less than b.

> _compare_ differs from <, >, <=, >=, == comparators in that it will

Reviewer: ```suggestion
_compare_ differs from `<`, `>`, `<=`, `>=`, `==`, and `!=` in that it will
```

> work for any type (e.g., `compare(1, "1")`). `null` values are the max value.

Reviewer: "Works for any type" is a little vague, but we can fix that later.
```suggestion
work for any type (e.g., `compare(1, "1")`). `null` values compare greater than non-`null` values.
```
